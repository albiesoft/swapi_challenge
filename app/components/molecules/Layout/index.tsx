import { FC } from 'react';
import { Layout } from 'antd';
import Loading from '../../atoms/Loading';
import Footer from '../../atoms/Footer';
import Header from '../../atoms/Header';

const MainLayout: FC = ({ children }) => {
  const { Content } = Layout;

  return (
    <Layout>
      <Loading>
        <Header />
        <Content style={{ padding: '40px 50px 0' }}>{children}</Content>
        <Footer />
      </Loading>
      <style jsx>{`
        min-height: calc(100vh - 139px);
      `}</style>
    </Layout>
  )
};

export default MainLayout;