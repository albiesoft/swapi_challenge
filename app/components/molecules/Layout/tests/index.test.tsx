import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import Layout from '..';

describe('<Loading />', () => {
  test('It renders correctly initially', () => {
    const target: ShallowWrapper = shallow(
      <Layout>
        <div></div>
      </Layout>
    );
    expect(target).toMatchSnapshot();
  });
});
