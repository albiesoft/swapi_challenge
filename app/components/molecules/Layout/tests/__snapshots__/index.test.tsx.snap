// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> It renders correctly initially 1`] = `
<Layout>
  <Loading>
    <Header />
    <Content
      style={
        Object {
          "padding": "40px 50px 0",
        }
      }
    >
      <div />
    </Content>
    <Footer />
  </Loading>
  <style>
    
        min-height: calc(100vh - 139px);
      
  </style>
</Layout>
`;
